lib/HTTP/BrowserDetect.pm

```diff
@@ -141,65 +206,91 @@ our @MISC_TESTS = qw(
-    lwp             => 'LWP::UserAgent',
-    lycos           => 'Lycos',
-    malware         => 'Malware / hack attempt',
-    mj12bot         => 'Majestic-12 DSearch',
-    msn             => 'MSN',
-    msnmobile       => 'MSN Mobile',
-    msoffice        => 'Microsoft Office',
-    nutch           => 'Apache Nutch',
-    phplib          => 'PHP http library',
-    puf             => 'puf',
-    robot           => 'robot',
-    rubylib         => 'Ruby http library',
-    slurp           => 'Yahoo! Slurp',
-    specialarchiver => 'archive.org_bot',
-    wget            => 'wget',
-    yahoo           => 'Yahoo',
-    yandex          => 'Yandex',
-    yandeximages    => 'YandexImages',
+    Applebot                             => 'Apple',
+    Discordbot                           => 'Discord',
+    'Google Page Speed'                  => 'Google Page Speed',
+    'Pro-Sitemaps'                       => 'Pro Sitemap Service',
+    Qwantify                             => 'Qwantify',
+    'SkypeUriPreview'                    => 'Skype URI Preview',
+    Swiftbot                             => 'Swiftbot',
+    W3C_Validator                        => 'W3C Validator',
+    WhatsApp                             => 'WhatsApp',
+    ahrefs                               => 'Ahrefs',
+    altavista                            => 'AltaVista',
+    apache                               => 'Apache http client',
+    askjeeves                            => 'AskJeeves',
+    baidu                                => 'Baidu Spider',
+    baiduspider                          => 'Baidu',
+    bingbot                              => 'Bingbot',
+    bitlybot                             => 'bitly',
+    curl                                 => 'curl',
+    'developers.google.com//web/snippet' => 'Google+ Snippet',
+    embedly                              => 'Embedly',
+    facebook                             => 'Facebook',
+    facebookexternalhit                  => 'Facebook',
+    flipboard                            => 'Flipboard',
+    getright                             => 'GetRight',
+    golib                                => 'Go language http library',
+    google                               => 'Googlebot',
+    googleadsbot                         => 'Google AdsBot',
+    googleadsense                        => 'Google AdSense',
+    googlebotimage                       => 'Googlebot Images',
+    googlebotnews                        => 'Googlebot News',
+    googlebotvideo                       => 'Googlebot Video',
+    googlefavicon                        => 'Google Favicon',
+    googlemobile                         => 'Googlebot Mobile',
+    indy                                 => 'Indy Library',
+    infoseek                             => 'InfoSeek',
+    ipsagent                             => 'Verisign ips-agent',
+    java                                 => 'Java',
+    lib                                  => 'lib',
```

Comment:
I worry about false positives with this.

Reply:
So, this was a tricky one, since the `lib()` method also denotes that a library is being used.  So, I removed `lib` from the robot regex, but added a workaround to preserve the method, even though I doubt it's used much, if at all.  But who knows what's still out there.